opty/direct_collocation.py

```diff
@@ -458,23 +513,51 @@ def plot_constraint_violations(self, vector, axes=None):
         con_nodes = range(1, self.collocator.num_collocation_nodes)
 
         plot_inst_viols = self.collocator.instance_constraints is not None
+        num_inst_viols = len(instance_violations)
+
         if axes is None:
-            fig, axes = plt.subplots(1 + plot_inst_viols, squeeze=False,
-                                     layout='compressed')
+            fig, axes = plt.subplots(1 + num_plots, squeeze=False,
+                layout='compressed', figsize=(8, 2.0*(num_plots+1)))
```

Comment:
It is better to leave the fig size for the user to create manually. Setting figsize inside a function like this can result in giant figures.